google/cloud/bigquery_storage_v1/reader.py

```diff
@@ -106,6 +106,12 @@ def __init__(self, wrapped, client, name, offset, read_rows_kwargs):
             read_rows_kwargs (dict):
                 Keyword arguments to use when reconnecting to a ReadRows
                 stream.
+            retry_delay_callback (Optional[Callable[[float], None]]):
```

Comment:
At first glance, I'm a bit confused why this parameter is necessary. Who needs a notification that we're going to sleep?

Reply:
Is it just for unit testing? In that case, please remove this parameter and use the `freezegun` library, instead.

Reply:
Users of the library may choose to provide this callback to be aware of delayed retries. When the users are aware of the delayed retry attempts, they can adjust their autoscaling algorithms. Apache Beam already does with the java sdk. My plan is to do the same with their python sdk.